scripts/busway-config-migrate.jq

```diff
@@ -0,0 +1,88 @@
+# The output is the entire updated configuration file.
+#
+# Usage:
+#   jq -f scripts/busway-config-migrate.jq priv/local.json > priv/new.json
+
+.screens += (
+  .screens | with_entries(
+    .value.app_params as $params
+    | .value.app_params.sections as $sections
+    | select(.value.app_id == "solari")
+    | select($params.overhead == false)
+    | {
+      key: "\(.key)-V2",
+      value: (.value + {
+        app_id: "busway_v2",
+        app_params: {
+          header: { stop_name: $params.station_name },
+          evergreen_content: [],
+          departures: {
+            sections: [
+              $sections[] | {
+                header: {
+                  title: (if .name == "" then null else .name end),
+                  arrow,
+                  read_as: (
+                    if .audio.wayfinding == .name then
+                      null
+                    else
+                      .audio.wayfinding
+                    end
+                  )
+                },
+                query,
+                filters: {
+                  max_minutes: .layout | (
+                    if .type == "bidirectional" or
+                        .opts.max_minutes == "infinity" then
+                      null
+                    else
+                      .opts.max_minutes
+                    end
+                  ),
+                  # NOTE: In the old filter format, `direction_id` must be only
+                  # one of the two directions; in the new format, it is allowed
+                  # to be `null`, indicating both directions. We should manually
+                  # combine instances of "route X dir 0" + "route X dir 1".
+                  route_directions: .layout | (
+                    if .type == "upcoming" and
+                        .opts.routes.route_list != [] then
+                      .opts.routes | {
+                        action,
+                        targets: .route_list[]
+                          | { route_id: .route, direction_id }
+                      }
+                    else
+                      null
+                    end
+                  )
+                },
+                layout: (
+                  if .layout.type == "upcoming" then
+                    # NOTE: The v1 config cannot specify a base and max size
+                    # separately; `visible_rows` (if Later Departures enabled)
+                    # or `num_rows` (if not) fills both roles. This sometimes
+                    # means extra space on the screen goes unused. We should
+                    # manually change `max` to `null` on sections where it
+                    # makes sense for them to use any extra space available.
+                    .layout.opts
+                    | (
+                        # `visible_rows` counts Later Departures as one "row".
+                        if .paged then .visible_rows - 1 else .num_rows end
```

Comment:
Does Later Departures get counted differently/separately in the new logic?

Reply:
The short answer is that it doesn't count at all. To quote myself in `ScreensConfig.V2.Departures.Layout`:

>   If `include_later` is set, the section includes a paging "Later Departures" component, which shows departures that would have otherwise been dropped due to space constraints (plus as many further-out departures as it has space for). The other fields do not "count" departures shown here; in other words, a section with `%{base: 4, max: 6, include_later: true}` is treated as desiring 4-6 departures that are visible at all times, and if more are available, however many will fit in "Later Departures" in addition to that.

Of course, so far this is only a description of the _intent_, as the section sizing logic has not been implemented yet.